flow/core/kernel/vehicle/traci.py

```diff
@@ -974,26 +974,31 @@ def get_x_by_id(self, veh_id):
         return self.master_kernel.network.get_x(
             self.get_edge(veh_id), self.get_position(veh_id))
 
-    def update_vehicle_colors(self):
+    def update_vehicle_colors(self, force_update):
         """See parent class.
 
         The colors of all vehicles are updated as follows:
         - red: autonomous (rl) vehicles
         - white: unobserved human-driven vehicles
         - cyan: observed human-driven vehicles
         """
+        if not self._color_vehicles:
+            return
+
         for veh_id in self.get_rl_ids():
             try:
-                # color rl vehicles red
-                self.set_color(veh_id=veh_id, color=RED)
+                if force_update or 'color' not in self.type_parameters[self.get_type(veh_id)]:
```

Comment:
Please add one comment explaining this line. It's the one thing that's confusing